reference.md

```diff
@@ -0,0 +1,381 @@
+
+## FinalizationRegistry Overview
+
+A `FinalizationRegistry` instance (a "registry") lets you get *cleanup callbacks* after objects registered with the registry are reclaimed. You create the registry passing in the callback:
+
+```js
+const registry = new FInalizationRegistry(heldValue => {
+    // ....
+});
+```
+
+Then you register any objects you want a cleanup callback for by calling the `register` method, passing in the object and a *held value* for it:
+
+```js
+registry.register(theObject, "some value");
+```
+
+The registry does not keep a strong reference to the object, as that would defeat the purpose (if the registry held it strongly, the object would never be reclaimed).
+
+If `theObject` is reclaimed, your cleanup callback may be called at some point in the future with the *held value* you provided for it (`"some value"` in the above). The held value can be any value you like: a primitive or an object, even `undefined`. If the held value is an object, the registry keeps a *strong* reference to it (so it can pass it to your cleanup callback later).
+
+If you might want to unregister an object later, you pass a third value, which is the *unregistration token* you'll use later when calling the registry's `unregister` function to unregister the object. The registry only keeps a weak reference to the unregister token.
+
+It's common to use the object itself as the unregister token, which is just fine:
+
+```js
+registry.register(theObject, "some value", theObject);
+// ...some time later, if you don't care about `theObject` anymore...
+registry.unregister(theObject);
+```
+
+It doesn't have to be the same object, though; it can be a different one:
+
+```js
+registry.register(theObject, "some value", tokenObject);
+// ...some time later, if you don't care about `theObject` anymore...
+registry.unregister(tokenObject);
+```
+
+## Notes on Cleanup Callbacks
+
+Developers shouldn't rely on cleanup callbacks for essential program logic. Cleanup callbacks may be useful for reducing memory usage across the course of a program, but are unlikely to be useful otherwise.
+
+A conforming JavaScript implementation, even one that does garbage collection, is not required to call cleanup callbacks. When and whether it does so is entirely down to the implementation of the JavaScript engine. When a registered object is reclaimed, the cleanup callbacks associated with any registries it's registered with may be called some time later, or not at all.
+
+It's likely that major implementations will call cleanup callbacks at some point during execution, but those calls may be substantially after the related object was reclaimed.
+
+There are also situations where even implementations that normally call cleanup callbacks are unlikely to call them:
+
+* When the JavaScript realm the objects existed in is being destroyed entirely (for instance, closing a tab in a browser).
```

Comment:
I think destruction is more of a per-agent thing than a per-realm thing. But let's just talk about this at a higher level, like, "When the JavaScript program shuts down entirely (for instance, ..."